lib/ansible/modules/files/assemble.py

```diff
@@ -148,7 +148,11 @@ def assemble_from_fragments(src_path, delimiter=None, compiled_regexp=None, igno
                 tmp.write(delimiter)
                 # always make sure there's a newline after the
                 # delimiter, so lines don't run together
-                if delimiter[-1] != b('\n'):
```

Comment:
Just as an FYI, you can do `delimiter[-1:]` to get the same behavior on both py2/py3.